includes/Modules/Analytics/Measurement_Code_Injector.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * Class Google\Site_Kit\Modules\Analytics\Measurement_Code_Injector
+ *
+ * @package   Google\Site_Kit\Modules\Analytics
+ * @copyright 2019 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Modules\Analytics;
+
+use Google\Site_Kit\Context;
+
+/**
+ * Injects Javascript based on the current active plugins
+ *
+ * Class Injector
+ */
+class Measurement_Code_Injector {
+	/**
+	 * A list of user's current active plugins that ShirshuClass supports
+	 *
+	 * @var array of string
+	 */
+	private $active_plugins = null;
+
+	/**
+	 * Holds a list of event configurations to be injected
+	 *
+	 * @var array
+	 */
+	private $event_configurations;
+
+	/**
+	 * Measurement_Event_Factory instance
+	 *
+	 * @var Measurement_Event_Factory
+	 */
+	private $event_factory = null;
+
+	/**
+	 * Plugin context.
+	 *
+	 * @since 1.0.0
+	 * @var Context
+	 */
+	protected $context;
+
+	/**
+	 * Injector constructor.
+	 *
+	 * @param array   $active_plugins list of supported plugins site currently has activated.
+	 * @param Context $module_context context used to check for AMP.
+	 */
+	public function __construct( $active_plugins, $module_context ) {
+		$this->active_plugins       = $active_plugins;
+		$this->context              = $module_context;
+		$this->event_factory        = Measurement_Event_Factory::get_instance();
+		$this->event_configurations = $this->build_event_configurations();
```

Comment:
This functionality only needs to run when the event tracking code actually gets injected, otherwise it is unnecessary, so for performance should be avoided. This could therefore be moved from the constructor into the `inject_event_tracking` method. Since this is the only place where it's used, we can also get rid of the class property.